crates/pallet-domains/src/lib.rs

```diff
@@ -1996,6 +2005,28 @@ impl<T: Config> Pallet<T> {
         })
     }
 
+    /// Returns the domain bundle limit of the given domain
+    pub fn domain_bundle_limit(
+        domain_id: DomainId,
+    ) -> Option<Result<DomainBundleLimit, ArithmeticError>> {
```

Comment:
A bit strange to see `Option<Result<..>>` how about changing to `Result<Option<..>>`?